pkg/server/api/agent/v1/service_test.go

```diff
@@ -24,10 +30,95 @@ func (c *serviceTest) Cleanup() {
 	c.done()
 }
 
+func TestCreateJoinToken(t *testing.T) {
+	for _, tt := range []struct {
+		name          string
+		request       *agentpb.CreateJoinTokenRequest
+		expectLogs    []spiretest.LogEntry
+		expectResults *types.JoinToken
+		err           string
+		code          codes.Code
+		dsError       error
+	}{
+		{
+			name: "Success Basic Create Join Token",
+			request: &agentpb.CreateJoinTokenRequest{
+				Ttl: 1000,
+			},
+		},
+		{
+			name: "Success Custom Value Join Token",
+			request: &agentpb.CreateJoinTokenRequest{
+				Ttl:   1000,
+				Token: "token goes here",
+			},
+		},
+		{
+			name: "Fail Negative Ttl",
+			request: &agentpb.CreateJoinTokenRequest{
+				Ttl: -1000,
+			},
+			err:  "ttl is required, you must provide one",
+			code: codes.InvalidArgument,
+		},
+		{
+			name: "Fail Datastore Error",
+			err:  "Failed to create token: datatore broken",
+			request: &agentpb.CreateJoinTokenRequest{
+				Ttl: 1000,
+			},
+			dsError: errors.New("datatore broken"),
+			code:    codes.Internal,
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test := setupServiceTest(t)
+			test.ds.SetNextError(tt.dsError)
+
+			result, err := test.client.CreateJoinToken(context.Background(), tt.request)
+			if tt.err != "" {
+				require.Error(t, err, tt.err)
+				spiretest.RequireGRPCStatusContains(t, err, tt.code, tt.err)
+			} else {
+				require.NoError(t, err)
```

Comment:
We have typically returned early from the `tt.err != ""` if instead of nesting the rest of the code under an `else`.